### PR TITLE
sieve: Document enotify extension

### PR DESCRIPTION
**Pull request by cmouse**
JIRA: DOV-4054

### CODE REVIEW

**Comment by cmouse** [comment]
Header/envelope tests [Sieve] together with Sieve variables can be used to
   extract the list of users to receive notifications from the incoming email
   message or its envelope. This is potentially quite dangerous, as this can be
   used for Deny Of Service attacks on recipients controlled by the message
   sender. For this reason implementations SHOULD NOT allow use of variables
   containing values extracted from the email message in the method parameter to
   the notify action.  Note that violation of this SHOULD NOT may result in* the
   creation of an open relay, i.e. any sender would be able to create specially
   crafted email messages that would result in notifications delivered to
   recipients under the control of the sender. In worst case this might result
   in financial loss by user controlling the Sieve script and/or by recipients
   of notifications (e.g. if a notification is an SMS message).

   --> This is currently not possible to check.

**Comment by stephanbosch** [comment]
> 
> 
> @stephanbosch What does this excerpt mean:
> 
> > Header/envelope tests [Sieve] together with Sieve variables can be used to
> > extract the list of users to receive notifications from the incoming email
> > message or its envelope. This is potentially quite dangerous, as this can be
> > used for Deny Of Service attacks on recipients controlled by the message
> > sender. For this reason implementations SHOULD NOT allow use of variables
> > containing values extracted from the email message in the method parameter to
> > the notify action.  Note that violation of this SHOULD NOT may result in* the
> > creation of an open relay, i.e. any sender would be able to create specially
> > crafted email messages that would result in notifications delivered to
> > recipients under the control of the sender. In worst case this might result
> > in financial loss by user controlling the Sieve script and/or by recipients
> > of notifications (e.g. if a notification is an SMS message).
> > --> This is currently not possible to check.

That is from the RFC. We currently do not check whether users are doing this.

File: source/configuration_manual/sieve/plugins/enotify.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
+.. _pigeonhole_plugin_enotify:
```
Comment (stephanbosch) [inline source/configuration_manual/sieve/plugins/enotify.rst:1]: Enotify is not a plugin

File: source/configuration_manual/sieve/plugins/enotify.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+(`RFC5435 <http://tools.ietf.org/html/rfc5435/>`__) adds notify action to the Sieve language.
+
+Configuration
+=============
+
+The plugin is activated by adding it to the :ref:`plugin-sieve-setting-sieve_plugins` setting:
+
+::
+
+   sieve_plugins = enotify
```
Comment (stephanbosch) [inline source/configuration_manual/sieve/plugins/enotify.rst:18]: Nope

File: source/configuration_manual/sieve/plugins/enotify.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+           # :matches is used to get the address from the From header
+           if address :matches :all "from" "*" {
+               set "from_addr" "${1}";
+           }
+
+           notify :message "${from_addr}${env_from}: ${subject}"
+                           "mailto:alm@example.com";
+       }
+
+Example using XMPP
```
Comment (stephanbosch) [inline source/configuration_manual/sieve/plugins/enotify.rst:88]: XMPP is not supported

**Review by stephanbosch — CHANGES_REQUESTED** [review]

File: source/configuration_manual/sieve/plugins/enotify.rst (hunk 1 above)
Comment (cmouse) [reply source/configuration_manual/sieve/plugins/enotify.rst:1]: so ... what is it? extension?

File: source/configuration_manual/sieve/plugins/enotify.rst (hunk 2 above)
Comment (cmouse) [reply source/configuration_manual/sieve/plugins/enotify.rst:18]: can you hint what it should say?

File: source/configuration_manual/sieve/plugins/enotify.rst (hunk 3 above)
Comment (cmouse) [reply source/configuration_manual/sieve/plugins/enotify.rst:88]: ok ... this was from RFC I think

File: source/configuration_manual/sieve/plugins/enotify.rst (hunk 1 above)
Comment (stephanbosch) [reply source/configuration_manual/sieve/plugins/enotify.rst:1]: yes

File: source/configuration_manual/sieve/plugins/enotify.rst (hunk 2 above)
Comment (stephanbosch) [reply source/configuration_manual/sieve/plugins/enotify.rst:18]: its not a plugin.. only an extension. so it cannot be added to sieve_plugins

File: source/configuration_manual/sieve/plugins/enotify.rst (hunk 3 above)
Comment (stephanbosch) [reply source/configuration_manual/sieve/plugins/enotify.rst:88]: yeah, they were working on the XMPP method in parallel and used it as an example. never got around to making enotify/xmpp support for Pigeonhole.

File: source/configuration_manual/sieve/extensions/enotify.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+The Sieve **enotify** extension
+(`RFC5435 <http://tools.ietf.org/html/rfc5435/>`__) adds notify action to the Sieve language.
+
+Configuration
+=============
+
+:ref:`plugin-sieve-setting-sieve_notify_mailto_envelope_from setting`
+   This allows configuring the source of the notification sender address for e-mail notifications.
+   This is similar to :ref:`plugin-sieve-setting-sieve_redirect_envelope_from` setting for redirect.
```
Comment (stephanbosch) [inline source/configuration_manual/sieve/extensions/enotify.rst:16]: This is similar to **the** sieve_redirect_envelope_from setting for redirect

File: source/configuration_manual/sieve/extensions/enotify.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+           }
+
+           notify :importance "3"
+               :message "[SIEVE] ${from}: ${subject}"
+               "mailto:alm@example.com";
+           fileinto "INBOX.sieve";
+       }
+
+Send notification is we receive mail from domain
+------------------------------------------------
```
Comment (stephanbosch) [inline source/configuration_manual/sieve/extensions/enotify.rst:54]: is?

File: source/settings/pigeonhole-ext/enotify.rst [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+.. _plugin-sieve-setting-sieve_notify_mailto_envelope_from setting:
+
+``sieve_notify_mailto_envelope_from setting``
+----------------------------------------------
+
+ - Default: <none>
+ - Value: :ref:`string`
+
+This allows configuring the source of the notification sender address for e-mail notifications.
+This is similar to :ref:`plugin-sieve-setting-sieve_redirect_envelope_from` setting for redirect.
```
Comment (stephanbosch) [inline source/settings/pigeonhole-ext/enotify.rst:16]: same +the comment

**Review by stephanbosch — CHANGES_REQUESTED** [review]

File: source/configuration_manual/sieve/extensions/enotify.rst [hunk 7]
```diff
@@ -0,0 +1,78 @@
+.. _pigeonhole_extension_enotify:
+
+=============================================
+Pigeonhole Sieve: Extension for Notifications
+=============================================
+
+
+The Sieve **enotify** extension
+(`RFC5435 <http://tools.ietf.org/html/rfc5435/>`__) adds notify action to the Sieve language.
```
Comment (stephanbosch) [inline source/configuration_manual/sieve/extensions/enotify.rst:9]: +the there somewhere

**Review by stephanbosch — CHANGES_REQUESTED** [review]

File: source/configuration_manual/sieve/extensions/enotify.rst (hunk 7 above)
Comment (cmouse) [reply source/configuration_manual/sieve/extensions/enotify.rst:9]: what?

Comment (stephanbosch) [reply source/configuration_manual/sieve/extensions/enotify.rst:9]: adds *the*/*a* notify action to the Sieve language.

**Comment by cmouse** [comment]
@stephanbosch fixed nits, can you review?

**Pull request merged**